orte/mca/ras/lsf/ras_lsf_module.c

```diff
@@ -112,8 +113,19 @@ static int allocate(orte_job_t *jdata, opal_list_t *nodes)
     /* release the nodelist from lsf */
     opal_argv_free(nodelist);
 
+    /* check to see if any mapping or binding directives were given */
+    if (NULL != jdata && NULL != jdata->map) {
+        if ((ORTE_MAPPING_GIVEN & ORTE_GET_MAPPING_DIRECTIVE(jdata->map->mapping)) ||
+            OPAL_BINDING_POLICY_IS_SET(jdata->map->binding)) {
+            directives_given = true;
+        }
+    } else if ((ORTE_MAPPING_GIVEN & ORTE_GET_MAPPING_DIRECTIVE(orte_rmaps_base.mapping)) ||
+               OPAL_BINDING_POLICY_IS_SET(opal_hwloc_binding_policy) {
```

Comment:
I believe a there is a missing parenthesis

Reply:
Quite correct - sorry about that. If you add one at the end of the `if` statement, does the revised code solve the problem?

Reply:
I also get a warning:
```
../../../../../orte/mca/ras/lsf/ras_lsf_module.c:122:65: error: ‘orte_rmaps_base’ undeclared (first use in this function); did you mean ‘orte_ras_base’?
```

Reply:
sigh - a missing include file. Please add:
```c
#include "orte/mca/rmaps/base/base.h"
```

Reply:
sorry for the problems - I don't have access to an LSF system and cannot compile that component.

Reply:
No worries, I'll try again

Reply:
now it got pass the ras_lsf_module.c

Here is my diff:
```bash
diff --git a/orte/mca/ras/lsf/ras_lsf_module.c b/orte/mca/ras/lsf/ras_lsf_module.c
index 43b3c7e757..ad3d8de2a2 100644
--- a/orte/mca/ras/lsf/ras_lsf_module.c
+++ b/orte/mca/ras/lsf/ras_lsf_module.c
@@ -34,6 +34,7 @@
 #include "opal/util/net.h"
 #include "opal/mca/hwloc/hwloc-internal.h"
 
+#include "orte/mca/rmaps/base/base.h"
 #include "orte/mca/rmaps/rmaps_types.h"
 #include "orte/mca/errmgr/errmgr.h"
 #include "orte/runtime/orte_globals.h"
@@ -120,7 +121,7 @@ static int allocate(orte_job_t *jdata, opal_list_t *nodes)
             directives_given = true;
         }
     } else if ((ORTE_MAPPING_GIVEN & ORTE_GET_MAPPING_DIRECTIVE(orte_rmaps_base.mapping)) ||
-               OPAL_BINDING_POLICY_IS_SET(opal_hwloc_binding_policy) {
+               OPAL_BINDING_POLICY_IS_SET(opal_hwloc_binding_policy)) {
             directives_given = true;
     }
 
```
I'll report back if it works :)

Reply:
It works, regardless of setting LSB_AFFINITY_HOSTFILE I get the requested mapping :)

Thanks!!! :)

Reply:
Great - thanks for the assist!